pkg/operator/controller/dns/controller_test.go

```diff
@@ -0,0 +1,119 @@
+package dns
+
+import (
+	"fmt"
+	"github.com/openshift/cluster-ingress-operator/pkg/dns"
+	"strings"
+
+	configv1 "github.com/openshift/api/config/v1"
+	operatorv1 "github.com/openshift/api/operator/v1"
+	iov1 "github.com/openshift/cluster-ingress-operator/pkg/api/v1"
+
+	"testing"
+)
+
+func TestPublishRecordToZones(t *testing.T) {
+	var tests = []struct {
+		name   string
+		record *iov1.DNSRecord
+		zones  []configv1.DNSZone
+		expect []operatorv1.ConditionStatus
+	}{
+		{
+			name: "testing for a successful update of 1 record",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: []operatorv1.ConditionStatus{operatorv1.ConditionFalse},
+		},
+		{
+			name: "when no zones available",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones:  []configv1.DNSZone{{}},
+			expect: []operatorv1.ConditionStatus{operatorv1.ConditionTrue},
+		},
+		{
+			name:   "when no record available",
+			record: nil,
+			zones: []configv1.DNSZone{{
+				ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+			},
+			expect: []operatorv1.ConditionStatus{operatorv1.ConditionTrue},
+		},
+		{
+			name: "when one zone available and one not available",
+			record: &iov1.DNSRecord{
+				Spec: iov1.DNSRecordSpec{
+					DNSName:    "subdomain.dnszone.io.",
+					RecordType: iov1.ARecordType,
+					Targets:    []string{"55.11.22.33"},
+				},
+			},
+			zones: []configv1.DNSZone{
+				{ID: "/subscriptions/E540B02D-5CCE-4D47-A13B-EB05A19D696E/resourceGroups/test-rg/providers/Microsoft.Network/dnszones/dnszone.io"},
+				{ID: ""}},
+			expect: []operatorv1.ConditionStatus{operatorv1.ConditionFalse, operatorv1.ConditionTrue},
+		},
+	}
+
+	var conditions []string
+	var expect []string
+
+	for _, test := range tests {
+		r := &reconciler{
+			dnsProvider: &FakeDNSProvider{},
+		}
+		t.Logf("evaluating test %s", test.name)
+		actual, results := r.publishRecordToZones(test.zones, test.record)
+		_ = results
+		for _, dnsStatus := range actual {
+			for _, condition := range dnsStatus.Conditions {
+				conditions = append(conditions, condition.Status)
+			}
+		}
+		for _, test_expect := range test.expect {
+			expect = append(expect, string(test_expect))
```

Comment:
Why do you need `expect`?  Why not use `test.expect` directly below?

Reply:
@Miciah actually test.expect has type []operatorv1.ConditionStatus  and condition has string data type so to compare I created string arrays for expected and conditions as different data type can't be compared

Reply:
Could you change the type of `expect` to `[]string` and use `reflect.DeepEqual` for the comparison?